### PR TITLE
add LinearRegressionOutput op to ngraph emitter

### PR DESCRIPTION
**Pull request by ashokei**
## Description ##
add LinearRegressionOutput op to ngraph emitter

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/ngraph/ngraph_compiler.h [hunk 1]
```diff
@@ -165,8 +165,9 @@ static std::unordered_set<std::string> ops_no_head_grad{
     "_greater_scalar",
     "_greater_equal_scalar",
     "_lesser_scalar",
-    "_lesser_equal_scalar",
-    "MakeLoss"};
+    "MakeLoss",
```
Comment (louisfeng) [inline src/ngraph/ngraph_compiler.h]: Would be nice if this list was ordered, easier to identify.

File: src/ngraph/ngraph_emitter.cc [hunk 2]
```diff
@@ -1552,6 +1555,16 @@ void Emitter::CreateLossOps() {
 
     return grad;
   };
+  loss_op_backward_funcs_["LinearRegressionOutput"] = [this](
+      const NodePtr& node, const NgraphNodePtr& adjoint) {
+    auto label = op_map_[node->inputs_[0]];
+    auto data = op_map_[node->inputs_[1]];
```
Comment (louisfeng) [inline src/ngraph/ngraph_emitter.cc]: One bug we hit for softmax is the data and label shape can be different. Could you verify this handles the case like https://github.com/NervanaSystems/ngraph-mxnet/blob/c02047ab46aaf2cc2a8ad2cb1c59c46eaa2d59ef/src/operator/regression_output-inl.h#L62

**Comment by ashokei** [comment]
@louisfeng added feedback to PR.
